src/ParametricOptInterface.jl

```diff
@@ -484,7 +484,17 @@ function MOI.set(
     c::MOI.ConstraintIndex,
     name::String,
 )
-    MOI.set(model.optimizer, attr, c, name)
+    if typeof(c).parameters[1] ==
```

Comment:
This is very misterious, I think you should parametrize the type to get the information that you want

Reply:
So, we should create three separate functions? Like for scalar quadratic function:
```julia
MOI.set(
    model::Optimizer,
    attr::MOI.ConstraintName,
    c::MOI.ConstraintIndex{F,S},
    name::String,
) where {S<:MOI.AbstractSet, F <: MOI.ScalarQuadraticFunction{T}}  where {T}
```
And so on...

Reply:
I thinnk that tis is a better approach than manually doing the runtime dispatch. We leave it to the julia compiler.